nxdrive/direct_edit.py

```diff
@@ -895,7 +912,11 @@ def _handle_upload_error(self, ref: Path, os_path: Path, /) -> None:
             "DIRECT_EDIT_UPLOAD_FAILED",
             [f'<a href="file:///{os_path.parent}">{ref.name}</a>'],
         )
+        remote.metrics.send(
+            {"NX-metric-drive.request": json.dumps(self._file_metrics[ref])}
+        )
         self._upload_errors.pop(ref, None)
+        self._file_metrics.pop(ref, None)
```

Comment:
```suggestion
        self._file_metrics.pop(ref)
```
You can apply the same change to the line above.